functions/src/index.ts

```diff
@@ -134,3 +129,14 @@ exports.initRepoPRs = firestore.document('repositories/{id}').onCreate((snapshot
     console.error(err);
   });
 });
+
+async function start() {
+  const accessToken = await getJwtToken(sizeAppConfig.clientEmail, sizeAppConfig.privateKey);
+
+  // Load the merge task to monitor PRs
+  bot.load(robot => {
+    tasks = registerTasks(robot, store, httpClient, accessToken, `https://${sizeAppConfig.projectId}.firebaseio.com`);
+  });
+}
+
+start();
```

Comment:
I don't like the idea of delaying the bot startup because we are waiting for the jwt token, you should only delay the size plugin. Maybe you should get that token in the constructor of the SizeTask instead. This way you don't have to change the way the apps are initialized in dev and index

Reply:
im now passing a ref to the Promise of getting the tokens, this allows for Unit testing but dosen't break the rest of the app if it dosent retrieve the token